themes/default/content/blog/pulumi-secrets-management/index.md

```diff
@@ -0,0 +1,200 @@
+
+```shell
+$ pulumi stack init passphrasestack --secrets-provider passphrase
+```
+
+or create a new project 
+```shell
+$ pulumi new azure-csharp --secrets-provider passphrase
+```
+
+It'll ask for the passphrase. This is the passphrase that Pulumi is going to use to encrypt the configuration system. ![Initialize the stack](./images/pulumi-stack-init-passphrase-1.jpg)
+
+Looking at the project in VSCode, we can see a new file has been created ![Create new file](./images/pulumi-stack-init-passphrase-2.jpg)
+
+Let's add two keys (one normal and one secret) 
+```sh
+$ pulumi config set normalKey normalValue` `pulumi config set --secret secretKey secretValue
+```
+![Add two keys](./images/pulumi-stack-init-passphrase-3.jpg)
+
+This is how our stack file looks like ![Stack output](./images/pulumi-stack-init-passphrase-4.jpg)
+
+If you pay attention, the secure value in `Pulumi.passphrasestack.yaml` is different to what we had in `Pulumi.dev.yaml`
+
+Notice, we need to enter the passphrase every time we are performing any operations. This can be avoided by setting `PULUMI_CONFIG_PASSPHRASE` in your environment variables (can be done on the local machine or in the CI system).
+
+### Using Azure Key Vault for encryption
+
+This would be a similar approach for any other cloud-based provider but we are looking at Azure Key Vault here. In order to use the Azure Key Vault for managing the encryption keys, the approach is same i.e., we set `--secret-provider` to Azure Key Vault via the CLI
+
+`sh
+$ pulumi stack init <name> --secrets-provider="<provider>://<provider-settings>"`
+
+e.g.,
+
+```sh
+$ pulumi stack init <name> --secrets-provider="azurekeyvault://<keyvaultname>.vault.azure.net/keys/<key name>"
+```
+
+Before we could do this, we need to create the Key Vault and a Key in the vault that we need.
+
+### Create Azure Key Vault
+
+```sh
+$ az keyvault create --name pulumiazvault --resource-group rg-pulumi-dev --location australiasoutheast
+```
+
+### Create the Key in Vault
+
+```sh
+$ az keyvault key create --name pulumiencryptionkey --vault-name pulumiazvault
+```
+
+### Assign Access
+
+When you create the vault and key from the CLI, a policy is created by default but it doesn't set 'encrypt' and 'decrypt' policy for us, so use this command to include that.
+
+Extract Object Id of your account (you can pull this from Azure AD or run the command below to extract it. Use your account here)\
+
+```sh
+$ az ad user show --id <your account name>
+```
+
+Look up for ObjectId of your account
+
+```sh
+$ az keyvault set-policy --name pulumiazvault --object-id <Object Id from above> --key-permissions encrypt decrypt get create delete list update import backup restore recover
+```
+
+*I prefer using Azure AD Application instead of any individual user account for such scenarios. In that case you will need the ObjectId of the application instead.*
```

Comment:
```suggestion
_I prefer using Azure AD Application instead of any individual user account for such scenarios. In that case you will need the ObjectId of the application instead._
```